simple_certmanager/forms.py

```diff
@@ -1,19 +1,95 @@
 from django import forms
+from django.core.files import File
+from django.utils.translation import gettext_lazy as _
 
 from .models import Certificate
+from .utils import (
+    BadPassword,
+    KeyIsEncrypted,
+    KeyIsNotEncrypted,
+    decrypted_key_to_pem,
+    load_pem_x509_private_key,
+)
+from .validators import PrivateKeyValidator
+
+
+def _read_and_reset(file_like: File):
+    """
+    Ensure that we read a file field from the start of the file and reset the pointer.
+    """
+    file_like.seek(0)
+    content = file_like.read()
+    file_like.seek(0)
```

Comment:
```suggestion
    org_offset = file_seek.tell()
    file_like.seek(0)
    content = file_like.read()
    file_like.seek(org_offset)
```

Would reset to where it was.

Reply:
we actually want to reset it to the beginning, since the file is consumed multiple times - once at the form level, and then again at the model field validation, so reads should always start at the beginning :)